public/Write-DbaDbTableData.ps1

```diff
@@ -728,7 +752,13 @@ function Write-DbaDbTableData {
             $bulkCopy.Dispose()
         }
 
-        # Close non-pooled connection as this is not done automatically. If it is a reused Server SMO, connection will be opened again automatically on next request.
-        $null = $server | Disconnect-DbaInstance
+        if (-not ($startedWithANonPooledConnection -or $usingGlobalTempTable) ) {
```

Comment:
It will leave the connection open and a strong assumption that the user knows they have to explicitly close it... which I don't like allowing.

I seem to recall this will cause issues for those that use this command with Azure SQL because it will cause all connections to be used up. That can adversely affect production apps if a one-off script is run against it.